lib/karafka/params/params.rb

```diff
@@ -40,30 +40,28 @@ class Params < Hash
         end
       end
 
-      # @return [Karafka::Params::Params] This method will trigger parser execution. If we decide
-      #   to retrieve data, parser will be executed to parse data. Output of parsing will be merged
-      #   to the current object. This object will be also marked as already parsed, so we won't
-      #   parse it again.
-      def parse!
-        return self if self['parsed']
+      # @return [Karafka::Params::Params] This method will trigger deserializer execution. If we
+      #   decide to retrieve data, deserializer will be executed to get data. Output of that will
+      #   be merged to the current object. This object will be also marked as already deserialized,
+      #   so we won't deserialize it again.
+      def deserialize!
+        return self if self['deserialized']
 
-        self['parsed'] = true
-        self['payload'] = parse(self['payload'])
+        self['deserialized'] = true
+        self['payload'] = deserialize(self['payload'])
         self
       end
 
       private
 
-      # @param payload [String] Raw data that we want to parse using consumer parser
-      # @note If something goes wrong, it will return raw data in a hash with a message key
-      # @return [Hash] parsed data or a hash with message key containing raw data if something
-      #   went wrong during parsing
-      def parse(payload)
-        Karafka.monitor.instrument('params.params.parse', caller: self) do
-          self['parser'].parse(payload)
+      # @param payload [String] Raw data that we want to deserialize using consumer deserializer
+      # @return [Object] deserialized data
+      def deserialize(payload)
+        Karafka.monitor.instrument('params.params.deserialize', caller: self) do
+          self['deserializer'].call(payload)
         end
-      rescue ::Karafka::Errors::ParserError => e
-        Karafka.monitor.instrument('params.params.parse.error', caller: self, error: e)
+      rescue ::StandardError => e
```

Comment:
I'm not sure about this, can't we catch a more specific error here?

Reply:
The question is, should require users to raise internal karafka error upon their deserializer execution or should we just allow it to raise whatever it wants and handle that. I would go with the second as it is less problematic for the end user, plus anyhow we don't have the "raw" data injection anymore from at leats `1.1`, so we reraise after the instrumentation.

Reply:
Sounds good